src/services/db/GitFileSystemService.ts

```diff
@@ -337,6 +338,86 @@ export default class GitFileSystemService {
     )
   }
 
+  /**
+   * Wrapper over `git diff --name-only` that also creates a local tracking branch for `base` and/or `head` if they do not exist.
+   * Note that `base` and `head` should be branch names, not commit hashes or other usually valid arguments for `git diff`.
+   */
+  getFilesChanged(
+    repoName: string,
+    base = "master",
+    head = "staging"
+  ): ResultAsync<string[], GitFileSystemError> {
+    if (base === head) {
+      return okAsync([])
+    }
+    return this.createLocalTrackingBranchIfNotExists(repoName, base)
+      .andThen(() => this.createLocalTrackingBranchIfNotExists(repoName, head))
+      .andThen(() => {
+        const efsVolPath = this.getEfsVolPathFromBranch(head)
+        return ResultAsync.fromPromise(
+          this.git
+            .cwd({ path: `${efsVolPath}/${repoName}`, root: false })
+            .diff([`${base}..${head}`, "--name-only"]),
+          (error) => {
+            logger.error(
+              `Error when getting diff files between "${base}" and "${head}": ${error}, when trying to access ${efsVolPath}/${repoName}`
+            )
+
+            if (error instanceof GitError) {
+              return new GitFileSystemError(
+                "Unable to retrieve git diff info from disk"
+              )
+            }
+
+            return new GitFileSystemError("An unknown error occurred")
+          }
+        ).map((files) =>
+          files
+            .trim()
+            .split("\n")
+            .filter((file) => file !== "")
```

Comment:
some of the files maybe `.keep`?

Reply:
hmm, I wanted to keep this method as generic as possible and let the caller handle specific edge cases like .keep. The `.keep` files are indeed filtered out in `ReviewRequestService.compareDiff` by checking for the `type !== "placeholder" (`type` is populated when constructing the `EditedItemInfo` for each file)

This was also the existing behaviour previously

Reply:
> hmm irrc we need the .keep, because in git it will not commit a folder if it is not empty folders!

not sure what you mean by this, but this method does not impact git commit activity, its only used when **reading** the files changed for the `/compare` and `/:requestId` endpoints

Reply:
oh no i agree with you @jacobkwan, we shouldnt filter out keep